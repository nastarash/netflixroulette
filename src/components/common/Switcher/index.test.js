import React from 'react';
import { Switcher } from './index';

describe('<Switcher />', () => {
  test('should be rendered correctly', () => {
    const variantNames = ['First', 'Second'];
    const switcher = mount(
        <Switcher variants={variantNames} default='First' />
    );
    expect(switcher).toMatchSnapshot();
  });

  test('button activity state is consistent', () => {
    const variantNames = ['One', 'Two', 'Three'];
    const switcher = shallow(
        <Switcher variants={variantNames} default='One' />
    );
    expect(switcher.childAt(0).childAt(0).prop('isActive')).toBeTruthy();
    expect(switcher.childAt(1).childAt(0).prop('isActive')).toBeFalsy();
    expect(switcher.childAt(2).childAt(0).prop('isActive')).toBeFalsy();
  });

  test('button activity state is changing correctly', () => {
    const variantNames = ['One', 'Two', 'Thr'];
    const switcher = mount(
        <Switcher variants={variantNames} default='One' />
    );

    expect(switcher.find('[text="One"]').prop('isActive')).toBeTruthy();
    expect(switcher.find('[text="Two"]').prop('isActive')).toBeFalsy();
    expect(switcher.find('[text="Thr"]').prop('isActive')).toBeFalsy();

    switcher.find('[text="Two"]').simulate('click');

    expect(switcher.find('[text="One"]').prop('isActive')).toBeFalsy();
    expect(switcher.find('[text="Two"]').prop('isActive')).toBeTruthy();
    expect(switcher.find('[text="Thr"]').prop('isActive')).toBeFalsy();
  });

  test('button click calling callback onChange() function', () => {
    const variantNames = ['One', 'Two', 'Thr'];
    const onChangeFn = jest.fn();
    const switcher = mount(
        <Switcher variants={variantNames} default='One' onChange={onChangeFn} />
    );
    switcher.find('[text="Two"]').simulate('click');
    expect(onChangeFn).toBeCalledWith('Two');
  });
});
