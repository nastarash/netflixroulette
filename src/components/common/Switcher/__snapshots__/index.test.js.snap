// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switcher /> should be rendered correctly 1`] = `
<Switcher
  default="First"
  variants={
    Array [
      "First",
      "Second",
    ]
  }
>
  <div
    className="switcher"
  >
    <div
      className="switcher__button"
      key="First"
    >
      <DummyButton
        isActive={true}
        onClick={[Function]}
        text="First"
        width="100%"
      >
        <input
          className="button button-active"
          onClick={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
          type="button"
          value="First"
        />
      </DummyButton>
    </div>
    <div
      className="switcher__button"
      key="Second"
    >
      <DummyButton
        isActive={false}
        onClick={[Function]}
        text="Second"
        width="100%"
      >
        <input
          className="button button-inactive"
          onClick={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
          type="button"
          value="Second"
        />
      </DummyButton>
    </div>
  </div>
</Switcher>
`;
